### ISSUE TITLE
Impossible to create CreditMemo | DivisionByZeroError: Division by zero in vendor/magento/module-sales/Model/Order/Creditmemo/Total/Tax.php:139

### ISSUE THREAD

**Issue by VladyslavSikailo** [magento/magento2#38053]
### Preconditions and environment

- Magento version: "2.4.6-p3"
- Sample Data installed

### Steps to reproduce

1. Configure 0 (zero) shipping price for any shipping method. For example for Flat Rate (`BO->Stores->Configuration->Sales->Delivery Methods->Flat Rate->Price`: 0)
2. Set `BO->Stores->Configuration->Sales->Tax->Orders, Invoices, Credit Memos Display Settings->Display Shipping Amount`: Including Tax
3. Go to frontend and place an order with any product using Flat Rate with 0 price
4. Go to admin, open placed order, create Invoice using admin
5. Go to order page, click on "Credit Memo" button to create Credit Memo

### Expected result

- New CreditMemo page should be opened
- it should be possible to created Credit Memo

### Actual result

Exception is thrown

```
DivisionByZeroError: Division by zero in /var/www/html/vendor/magento/module-sales/Model/Order/Creditmemo/Total/Tax.php:139
Stack trace:
#0 /var/www/html/vendor/magento/module-sales/Model/Order/Creditmemo.php(374): Magento\Sales\Model\Order\Creditmemo\Total\Tax->collect(Object(Magento\Sales\Model\Order\Creditmemo))
#1 /var/www/html/vendor/magento/module-sales/Model/Order/CreditmemoFactory.php(99): Magento\Sales\Model\Order\Creditmemo->collectTotals()
#2 /var/www/html/vendor/magento/module-sales/Controller/Adminhtml/Order/CreditmemoLoader.php(220): Magento\Sales\Model\Order\CreditmemoFactory->createByOrder(Object(Magento\Sales\Model\Order\Interceptor), Array)
#3 /var/www/html/vendor/magento/module-sales/Controller/Adminhtml/Order/Creditmemo/NewAction.php(64): Magento\Sales\Controller\Adminhtml\Order\CreditmemoLoader->load()
#4 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction->execute()
#5 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->___callParent('execute', Array)
#6 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->Magento\Framework\Interception\{closure}()
#7 /var/www/html/generated/code/Magento/Sales/Controller/Adminhtml/Order/Creditmemo/NewAction/Interceptor.php(23): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->___callPlugins('execute', Array, Array)
#8 /var/www/html/vendor/magento/framework/App/Action/Action.php(111): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->execute()
#9 /var/www/html/vendor/magento/module-backend/App/AbstractAction.php(151): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#10 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#11 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->___callParent('dispatch', Array)
#12 /var/www/html/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(145): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#13 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(135): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#14 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#15 /var/www/html/generated/code/Magento/Sales/Controller/Adminhtml/Order/Creditmemo/NewAction/Interceptor.php(32): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->___callPlugins('dispatch', Array, Array)
#16 /var/www/html/vendor/magento/framework/App/FrontController.php(245): Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#17 /var/www/html/vendor/magento/framework/App/FrontController.php(212): Magento\Framework\App\FrontController->getActionResponse(Object(Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor), Object(Magento\Framework\App\Request\Http))
#18 /var/www/html/vendor/magento/framework/App/FrontController.php(147): Magento\Framework\App\FrontController->processRequest(Object(Magento\Framework\App\Request\Http), Object(Magento\Sales\Controller\Adminhtml\Order\Creditmemo\NewAction\Interceptor))
#19 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#20 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#21 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#22 /var/www/html/generated/code/Magento/Framework/App/FrontController/Interceptor.php(23): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#23 /var/www/html/vendor/magento/framework/App/Http.php(116): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#24 /var/www/html/generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http->launch()
#25 /var/www/html/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#26 /var/www/html/pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#27 {main}
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by ayushi011** [comment]
Hello @VladyslavSikailo,

Adobe has already released a patch for this issue. This issue will be fixed in magento version 2.4.7. Please refer below patch - ACSD-50814: Admin user is not able to create credit memo

Link- https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/support-tools/patches/v1-1-30/acsd-50814-an-admin-user-is-not-able-to-create-credit-memo.html?lang=en
https://github.com/magento/quality-patches/blob/b0065b97798f98f5f46a2ab2be8463f08cf01f06/patches/os/ACSD-50814_2.4.6.patch

Thanks,

**Comment by engcom-Bravo** [comment]
@ayushi011 Thanks for your contribution & collaboration over here.

Hi @VladyslavSikailo,

Thank you for reporting and collaboration.

Adobe released a quality patch for this https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/support-tools/patches/v1-1-30/acsd-50814-an-admin-user-is-not-able-to-create-credit-memo.html?lang=en

**Comment by engcom-Bravo** [comment]
@ayushi011  Thanks for your contribution & collaboration over here

Hi @VladyslavSikailo,

Thank you for reporting and collaboration.

Adobe released a quality patch for this https://github.com/magento/quality-patches/blob/b0065b97798f98f5f46a2ab2be8463f08cf01f06/patches/os/ACSD-50814_2.4.6.patch  

with the following https://experienceleague.adobe.com/docs/commerce-knowledge-base/kb/support-tools/patches/v1-1-30/acsd-50814-an-admin-user-is-not-able-to-create-credit-memo.html?lang=en

Please refer the above links the issue has been fixed in Magento 2.4 develop.

Hence we are closing this issue.

Thanks.

**Comment by VladyslavSikailo** [comment]
Thank you guys!
